src/base/rss/rss_feed.cpp

```diff
@@ -0,0 +1,418 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2015, 2017  Vladimir Golovnev <glassez@yandex.ru>
+ * Copyright (C) 2010  Christophe Dumez <chris@qbittorrent.org>
+ * Copyright (C) 2010  Arnaud Demaiziere <arnaud@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "rss_feed.h"
+
+#include <QCryptographicHash>
+#include <QDir>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QScopedPointer>
+#include <QUrl>
+
+#include "../asyncfilestorage.h"
+#include "../logger.h"
+#include "../net/downloadhandler.h"
+#include "../net/downloadmanager.h"
+#include "../profile.h"
+#include "../utils/fs.h"
+#include "private/rss_parser.h"
+#include "rss_article.h"
+#include "rss_session.h"
+
+const QString DataFolderName(QStringLiteral("rss/data"));
+
+const QString Str_Url(QStringLiteral("url"));
+const QString Str_Title(QStringLiteral("title"));
+const QString Str_LastBuildDate(QStringLiteral("lastBuildDate"));
+const QString Str_IsLoading(QStringLiteral("isLoading"));
+const QString Str_HasError(QStringLiteral("hasError"));
+const QString Str_Articles(QStringLiteral("articles"));
+
+using namespace RSS;
+
+Feed::Feed(const QString &url, const QString &path, Session *session)
+    : Item(path)
+    , m_session(session)
+    , m_url(url)
+{
+    m_dataFileName = QString("%1.json").arg(
+                QString(QCryptographicHash::hash(m_url.toUtf8(), QCryptographicHash::Sha1).toHex()));
```

Comment:
Why do you need human friendly name for these files? It's not for user editing (and even reading in most cases). Are you sure you really want it?